x-pack/legacy/plugins/integrations_manager/server/integrations/data.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  SavedObject,
+  SavedObjectsBulkGetObject,
+  SavedObjectsClientContract,
+} from 'src/core/server/saved_objects';
+import { InstallationSavedObject, Installable } from '../../common/types';
+import { SAVED_OBJECT_TYPE } from '../../common/constants';
+import * as Registry from '../registry';
+
+export async function getIntegrations(client: SavedObjectsClientContract) {
+  const registryItems = await Registry.fetchList();
+  const searchObjects: SavedObjectsBulkGetObject[] = registryItems.map(({ name, version }) => ({
+    type: SAVED_OBJECT_TYPE,
+    id: `${name}-${version}`,
+  }));
+
+  const results = await client.bulkGet(searchObjects);
+  const savedObjects: InstallationSavedObject[] = results.saved_objects.filter(o => !o.error); // ignore errors for now
+  const integrationList = registryItems
+    .map(item =>
+      createInstallableFrom(
+        item,
+        savedObjects.find(({ id }) => id === `${item.name}-${item.version}`)
+      )
+    )
+    .sort(sortByName);
+
+  return integrationList;
+}
+
+export async function getIntegrationInfo(client: SavedObjectsClientContract, pkgkey: string) {
```

Comment:
Is this the single resource version of `getIntegrations`?

Reply:
Yes, IIUC. It’s what’s used for the detail page. `getIntegrations` is what’s used for for the list view.